ttnn/cpp/ttnn/tensor/layout/tensor_layout.hpp

```diff
@@ -51,6 +51,15 @@ class TensorLayout {
     //  H is all dimensions except W multiplied and aligned to tile and shard height
     Size compute_physical_shape(const ttnn::SimpleShape& shape) const;
 
+    void set_memory_config(MemoryConfig memory_config) {
```

Comment:
wouldn't want it to be mutable like this. lets discuss

Reply:
I agree. I needed to create exactly the same TensorLayout, but with a different memory config. I didn't figure out a good way to do it. I can create a method `TensorLayout with_memory_config(MemoryConfig) const;`, what do you think?